tests/FsCheck.Test/Gen.fs

```diff
@@ -119,6 +119,12 @@ module Gen =
         Gen.listOfLength length (Gen.constant v)
         |> sample1
         |> ((=) (List.init length (fun _ -> v)))
+
+    [<Property>]
+    let Shuffle (NonEmptySet (xs:Set<int>)) =
+        Gen.shuffle xs
+        |> sample 10
+        |> List.exists ((<>) (Set.toSeq xs))
```

Comment:
No need for RNG. Just use a standard Shuffle algorithm, like [Fisher-Yates](https://en.wikipedia.org/wiki/Fisher%E2%80%93Yates_shuffle)

I implemented an example here:

https://github.com/Aaronontheweb/faker-csharp/blob/v2.0/Faker/Helpers/ArrayHelpers.cs#L61

Model tests for said function: shuffled order should never match original https://github.com/Aaronontheweb/faker-csharp/blob/v2.0/Faker.Models.Tests/ArrayHelperSpecs.cs#L12

I also wrote a composite test that verifies that the shuffled sequence always contains the same original elements

https://github.com/Aaronontheweb/faker-csharp/blob/v2.0/Faker.Models.Tests/TypeTableSpecs.cs#L62